File: sn-hd-controler/src/main/webapp/service/leaderService.js
//服务层
app.service('leaderService',function($http){

    //读取列表数据绑定到表单中
    this.findAll=function(){
        return $http.get('../leader/findAll.do');
    }
    /**
     * 分页展示数据
     * @param page
     * @param rows
     */
    this.findPage=function(page,rows){
        return $http.get('../leader/findPage.do?page='+page+'&rows='+rows);
    }
    /**
     * 根据id查询实体
     * @param id
     */
    this.findOne=function(id){
        return $http.get('../leader/findOne.do?id='+id);
    }
    /**
     * 添加实体数据
     * @param entity
     */
    this.add=function(entity){
        return  $http.post('../leader/add.do',entity );
    }
    /**
     * 修改数据
     * @param entity
     */
    this.update=function(entity){
        return  $http.post('../leader/update.do',entity );
    }

    /**
     * 根据id删除数据或批量删除数据
     * @param ids
     */
    this.delete=function(ids){
        return $http.get('../leader/delete.do?ids='+ids);
    }
    /**
     * 搜索数据
     * @param page
     * @param rows
     * @param searchEntity
     */
    //搜索
    this.search=function(page,rows,searchEntity){
        return $http.post('../leader/search.do?page='+page+"&rows="+rows, searchEntity);
    }
    /**
     * 同步数据到solr
     */
    this.updateSolr=function(){
        return $http.post('../leader/solrUpdate.do');
    }
});

File: sn-search-interface/src/main/java/com/sn/search/LeaderService.java
package com.sn.search;

import com.sn.entity.PageResult;
import com.sn.pojo.Leader;

import java.util.List;

public interface LeaderService {

    /**
     * 返回全部列表
     * @return
     */
    public List<Leader> findAll();


    /**
     * 返回分页列表
     * @return
     */
    public PageResult findPage(int pageNum, int pageSize);


    /**
     * 增加
     */
    public void add(Leader leader);


    /**
     * 修改
     */
    public void update(Leader leader);


    /**
     * 根据ID获取实体
     * @param id
     * @return
     */
    public Leader findOne(Long id);


    /**
     * 批量删除
     * @param ids
     */
    public void delete(Long[] ids);

    /**
     * 更新solr数据
     */
    void updateSOlr();

    /**
     * 分页
     * @param pageNum 当前页 码
     * @param pageSize 每页记录数
     * @return
     */
    public PageResult findPage(Leader leader, int pageNum, int pageSize);


}
